test/SlowTests/Client/Queries/RavenDB_14541.cs

```diff
@@ -0,0 +1,394 @@
+                var company2 = new Company { Name = "App" };
+                var company3 = new Company { Name = "Micro" };
+                var company4 = new Company { Name = "Amaz" };
+
+                session.Store(employee1);
+                session.Store(employee2);
+                session.Store(company1, "Companies/Raven");
+                session.Store(company2, "Companies/App");
+                session.Store(company3, "Companies/Micro");
+                session.Store(company4, "Companies/Amaz");
+
+                session.SaveChanges();
+            }
+
+            new Employees_ByFirstName().Execute(store);
+            Indexes.WaitForIndexing(store);
+        }
+
+        [RavenFact(RavenTestCategory.Querying)]
+        public void ShouldThrowInvalidOperationExceptionHaveIncludesInsteadOf_()
+        {
+            using (DocumentStore store = GetDocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new Order { Company = "1", Employee = "employees/1" }, "orders/1-A");
+                    session.Store(new Order { Company = "2", Employee = "employees/2" }, "orders/2-A");
+
+                    session.Store(new Employee { FirstName = "a" }, "employees/1");
+                    session.Store(new Employee { FirstName = "b" }, "employees/2");
+
+                    session.SaveChanges();
+                }
+
+                Assert.Throws<InvalidOperationException>(() =>
+                {
+                    using (var session = store.OpenSession())
+                    {
+                        var query3 = from o in session.Query<Order>()
+                            let includes = RavenQuery.Include<Order>(u => u.Employee)
+                            select new QueryResult { Comapny = o.Company };
+
+                        var results = query3.ToList();
+                    }
+                });
+            }
+        }
+
+        [RavenFact(RavenTestCategory.Querying)]
+        public void SessionQuerySelectAddressFromIncludeDoc_UsingRavenQueryStringWithStateObject()
+        {
+            using (DocumentStore store = GetDocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new Address { CountryState = "states/1#zip07", City = "new-york", StateId = "states/1" });
+                    session.Store(new Address { CountryState = "states/2#zip05", City = "haifa", StateId = "states/2" });
+
+                    session.Store(new State { Name = "Alabama" }, "states/1");
+                    session.Store(new State { Name = "Minassota" }, "states/2");
+
+                    session.SaveChanges();
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var query3 = from a in session.Query<Address>()
+                        let _ = RavenQuery.Include<Address>(x => x.StateId)
```

Comment:
I didn’t add support for inferring the type from the a parameter because the other methods inside the RavenQuery class follow the same approach and don’t implement it either. I wanted to maintain consistency across the class.

However, if you’d prefer me to implement this, I’d be happy to add support for it. 
Let me know how you’d like me to proceed.

Reply:
Let's keep it as is for now, thx